Dockerfile

```diff
@@ -1,4 +1,46 @@
-FROM ubuntu:18.04
+FROM ubuntu:18.04 as builder
+LABEL maintainer="info@camptocamp.com"
+
+RUN apt-get update && \
+    LC_ALL=C DEBIAN_FRONTEND=noninteractive apt-get install -y bison flex python-lxml libfribidi-dev swig \
+        cmake librsvg2-dev colordiff libpq-dev libpng-dev libjpeg-dev libgif-dev libgeos-dev libgd-dev \
+        libfreetype6-dev libfcgi-dev libcurl4-gnutls-dev libcairo2-dev libgdal-dev libproj-dev libxml2-dev \
+        libxslt1-dev python-dev php-dev libexempi-dev lcov lftp libgdal-dev ninja-build git curl ccache \
+        clang libprotobuf-c-dev protobuf-c-compiler libharfbuzz-dev libcairo2-dev librsvg2-dev && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+ARG MAPSERVER_BRANCH
+
+RUN git clone https://github.com/mapserver/mapserver --branch=${MAPSERVER_BRANCH} --depth=100 /src
+
+COPY checkout_release /tmp
+RUN cd /src; /tmp/checkout_release ${MAPSERVER_BRANCH}
+
+ENV \
+    CXX="/usr/lib/ccache/clang++" \
+    CC="/usr/lib/ccache/clang"
```

Comment:
I'd say that ccache is useless in a multi-stage build context. You should remove it everywhere...

Reply:
+1, thanks :-)